### ISSUE TITLE
Inconsistent results of concordance function

### ISSUE THREAD

**Issue by ScottZimmerman** [therneau/survival#189]
We're using the concordance function on a training data set (n ~ 200,000) and evaluating concordance on a test set (n ~ 100,000). When running concordance with "newdata" we don't always get the same result, even after setting a seed.

The influence matrix returned by concordance when the newdata option is specified sometimes contains negative and/or non-integer and/or impossibly large or small values and/or NaNs (since these are counts, this should never happen, and counts should be bounded by the possible number of comparisons in the range [0,(n^2 - n)/2)]. 

This bug seems to occur randomly, and I suspect there is a memory allocation problem in the [C subroutine](https://github.com/cran/survival/blob/3ad71a6337e3735b83356a46d2ff5a292984cdb8/src/survConcordance.c), although I haven't been able to check that yet.

**Example code**
set.seed(18)
fitimp <- survival::coxph(formula, data=d_train)
set.seed(54)
test<- concordance(fitimp, newdata = d_test, influence=2, ranks=TRUE)
View(test$influence)

**Correct output**
<img width="419" alt="thumbnail_Screen Shot 2022-03-28 at 1 37 48 PM" src="https://user-images.githubusercontent.com/8963903/160916182-9d321fc8-71f1-4cfc-a24f-bfce67c433b8.png">

**One example of bugged output**
<img width="448" alt="thumbnail_Screen Shot 2022-03-28 at 1 35 55 PM" src="https://user-images.githubusercontent.com/8963903/160916197-ca1a7ff4-5c13-4cd2-a87e-b75dd3cf51b1.png">

**Comment by therneau** [comment]
I'm not quite clear on this
 1. random numbers don't play a role in the concordance function.  
 2. Your example code doesn't tell me anything concrete.   What is the formula?  What is d_train and/or d_test? 
 3. What exactly are you printing out in the output?

A bit more to go on would help me dig into this.
Terry

**Comment by therneau** [comment]
I have managed to recreate the problem, and it was yet another facet of the rounding issues discussed in the "tied times" vignette.  The use of newdata gave a path through the routine that skipped the aeqSurv checks; while the use of a large simulated data set almost guaranteed that there would be response times that differ by less than machine precision.    I've now plugged that gap.
